compiler/src/iree/compiler/DispatchCreation/SetSplitReductionSizes.cpp

```diff
@@ -271,36 +270,34 @@ struct SetSplitReductionSizesPass final
     int64_t depthSize = getSizeAt(outputShape, depthPos);
 
     // The constants below are determined based on empirical data.
-    const int64_t largeDimSize = 512;
-    const int64_t mediumDimSize = 128;
-    const int64_t smallDimSize = 32;
+    const int64_t largeParallelSize = 512;
+    const int64_t largeReductionSize = 8192;
+    const int64_t ratioThreshold = 64;
 
     // When the batch and output channel sizes are large, the workload tends
     // to distributed across many workgroups, making split reduction little to
     // no effect.
-    if (outputChannelSize >= largeDimSize && batchSize >= largeDimSize) {
+    if (outputChannelSize >= largeParallelSize &&
+        batchSize >= largeParallelSize) {
       LDBG() << "skipping op; large output channel or batch size";
       return std::nullopt;
     }
 
-    // When the input spatial sizes are small while the batch and output channel
-    // sizes are relatively larger, split reduction often has no effect or even
-    // degrades performance.
-    for (auto dim : convDims->filterLoop) {
-      for (auto [idx, e] : llvm::enumerate(inputMap.getResults())) {
-        if (e.isFunctionOfDim(dim) && inputShape[idx] < smallDimSize &&
-            outputChannelSize > mediumDimSize && batchSize > mediumDimSize) {
-          LDBG() << "skipping op; small input spatial size";
-          return std::nullopt;
-        }
-      }
+    // When the reduction size is small relative to the output sizes, split
+    // reduction often has no effect or even degrades performance.
+    SmallVector<int64_t> tileSizes = std::move(*maybeSizes);
+    int64_t reductionSize = std::accumulate(tileSizes.begin(), tileSizes.end(),
+                                            1, std::multiplies<>());
```

Comment:
This usually has bugs, because the implementation can use int type for accumulator. The prefer way is `llvm::product_of(tileSizes)`.